components/sup/src/manager/service/config.rs

```diff
@@ -81,12 +82,17 @@ impl ServiceConfig {
     pub fn new(package: &PackageInstall,
                mgr_cfg: &ManagerConfig,
                runtime_cfg: &RuntimeConfig,
-               config_root: Option<PathBuf>,
+               config_root_from_config: Option<PathBuf>,
                bindings: Vec<(String, ServiceGroup)>)
                -> Result<ServiceConfig> {
-        let config_root = config_root.unwrap_or(package.installed_path.clone());
+        let config_root = match config_root_from_config {
```

Comment:
because we are owning `config_root_from_config` and assigning it to our own property below, The replaced line had to clone `config_root_from_config `. and it would need to clone regardless of whether it would use it (like if it was `None`) so I broke it out to only clone if it was used. Maybe its just a micro optimization since cloning `None` is little to none.